src/main/java/au/com/southsky/jfreesane/SaneDevice.java

```diff
@@ -214,4 +217,41 @@ SaneSession getSession() {
   void invalidateOptions() {
     optionTitleMap = null;
   }
+
+  @Override
+  public int hashCode() {
+    int hash = 3;
+    hash = 37 * hash + Objects.hashCode(this.name);
+    hash = 37 * hash + Objects.hashCode(this.vendor);
+    hash = 37 * hash + Objects.hashCode(this.model);
+    hash = 37 * hash + Objects.hashCode(this.type);
+    return hash;
+  }
+
+  @Override
+  public boolean equals(Object obj) {
```

Comment:
> Well, you'll recreate (most) of the state on the client, but you need to tell the server as well. JFreeSane only sets option values upon explicit calls to set*Value, not unconditionally at the beginning of the scan.

That sounds straightforward and easy when it's well documented. Maybe `SaneSession` can get a `reattachDevice` or `reconnectDevice` method doing what `SaneSession.withRemoteSane` does and write options of `SaneDevice`/`SaneDeviceOptions` on the server (or just have a `SaneSession.withRemoteSane` with a `SaneDevice reconnectDevice` argument)?